templates/avm/examples/config-avm-hub-and-spoke-vnet-multi-region.tfvars

```diff
@@ -0,0 +1,270 @@
+`subscription_id_management`: The subscription ID of the subscription to deploy the management resources to, sourced from the variable `subscription_id_management`.
+*/
+
+/* 
+Custom Names: Based on any of the above variables, you can create your own custom names to use in the configuration by supplying them in the `custom_names` map variable.
+This avoids repeating the same hard-coded values in the configuration. 
+For example, you can use the custom name `$${management_resource_group_name}` in the configuration instead of hard-coding the value `rg-management-$${starter_location_01}`.
+NOTE: You cannot build a custom name based on another custom name. You can only build a custom name based on the templated variables.
+*/
+custom_names = {
+  # Resource group names
+  management_resource_group_name = "rg-management-$${starter_location_01}"
+
+  connectivity_hub_primary_resource_group_name   = "rg-hub-$${starter_location_01}"
+  connectivity_hub_secondary_resource_group_name = "rg-hub-$${starter_location_02}"
+  dns_resource_group_name                        = "rg-hub-dns-$${starter_location_01}"
+  ddos_resource_group_name                       = "rg-hub-ddos-$${starter_location_01}"
+
+  # Resource names
+  log_analytics_workspace_name            = "law-management-$${starter_location_01}"
+  ddos_protection_plan_name               = "ddos-hub-$${starter_location_01}"
+  automation_account_name                 = "aa-management-$${starter_location_01}"
+  ama_user_assigned_managed_identity_name = "uami-management-ama-$${starter_location_01}"
+  dcr_change_tracking_name                = "dcr-change-tracking"
+  dcr_defender_sql_name                   = "dcr-defender-sql"
+  dcr_vm_insights_name                    = "dcr-vm-insights"
+
+  # Resource identifiers
+  management_resource_group_id = "/subscriptions/$${subscription_id_management}/resourcegroups/rg-management-$${starter_location_01}"
+}
+
+management_resource_settings = {
+  automation_account_name      = "$${automation_account_name}"
+  location                     = "$${starter_location_01}"
+  log_analytics_workspace_name = "$${log_analytics_workspace_name}"
+  resource_group_name          = "$${management_resource_group_name}"
+  user_assigned_managed_identities = {
+    ama = {
+      name = "$${ama_user_assigned_managed_identity_name}"
+    }
+  }
+  data_collection_rules = {
+    change_tracking = {
+      name = "$${dcr_change_tracking_name}"
+    }
+    defender_sql = {
+      name = "$${dcr_defender_sql_name}"
+    }
+    vm_insights = {
+      name = "$${dcr_vm_insights_name}"
+    }
+  }
+}
+
+management_group_settings = {
+  location           = "$${starter_location_01}"
+  architecture_name  = "alz"
+  parent_resource_id = "$${root_parent_management_group_id}"
+  policy_default_values = {
+    ama_change_tracking_data_collection_rule_id = "$${management_resource_group_id}/providers/Microsoft.Insights/dataCollectionRules/$${dcr_change_tracking_name}"
```

Comment:
can we use provider functions here, is that any better?

Reply:
I'm not sure if it will help here as I'm trying to provide unlimited flexibility to people using the config file. Using the provider function may be possible, but not inline in the tfvars file, so would involve a new object type.

I appreciate there is scope for human error here, but I think the people delivering the VBD would prefer this level for flexibility.